samplesheets/views.py

```diff
@@ -885,6 +909,191 @@ def handle_alerts_deactivate(cls, irods_request, app_alerts=None):
+                message='iRODS delete request accepted by {}: "{}"'.format(
+                    request.user.username, irods_request.get_short_path()
+                ),
+                level='SUCCESS',
+                url=reverse(
+                    'samplesheets:project_sheets',
+                    kwargs={'project': project.sodar_uuid},
+                ),
+                project=project,
+            )
+            # Handle project alerts
+            cls.handle_alerts_deactivate(irods_request, app_alerts)
+
+    @classmethod
+    def reject_request(
+        cls, irods_request, project, request, timeline=None, app_alerts=None
+    ):
+        """
+        Reject an iRODS delete request.
+
+        :param irods_request: IrodsDataRequest object
+        :param project: Project object
+        :param request: Request object
+        :param timeline: Timeline API or None
+        :param app_alerts: Appalerts API or None
+        """
+        if timeline:
+            tl_event = timeline.add_event(
+                project=project,
+                app_name=APP_NAME,
+                user=request.user,
+                event_name='irods_request_reject',
+                description='reject data iRODS request {irods_request}',
+                status_type='OK',
+            )
+            tl_event.add_object(
+                obj=irods_request,
+                label='irods_request',
+                name=irods_request.get_display_name(),
+            )
+
+        # Prepare and send notification email
+        if (
+            settings.PROJECTROLES_SEND_EMAIL
+            and irods_request.user != request.user
+        ):
+            subject_body = 'iRODS delete request rejected'
+            message_body = EMAIL_DELETE_REQUEST_REJECT.format(
+                project=irods_request.project.title,
+                user=irods_request.user.username,
+                user_email=irods_request.user.email,
+                path=irods_request.path,
+            )
+            send_generic_mail(
+                subject_body, message_body, [irods_request.user.email], request
+            )
+
+        # Create app alert
+        if app_alerts and irods_request.user != request.user:
+            app_alerts.add_alert(
+                app_name=APP_NAME,
+                alert_name=IRODS_REQ_REJECT_ALERT,
+                user=irods_request.user,
+                message='iRODS delete request rejected by {}: "{}"'.format(
+                    request.user.username, irods_request.get_short_path()
+                ),
+                level='WARNING',
+                url=reverse(
+                    'samplesheets:project_sheets',
+                    kwargs={'project': project.sodar_uuid},
+                ),
+                project=project,
+            )
+            # Handle project alerts
+            cls.handle_alerts_deactivate(irods_request, app_alerts)
+
+    def get_zone_permissions(self, request, zone):
```

Comment:
I think I was previously confused by two different pull requests :D

Of course this has nothing to do with any landing zones at all, but with irods data requests. Hence this should be called something like  `has_irods_request_perms()` and the object is `irods_request`, not `zone`.

Sorry for being confusing. I need a vacation..

Eventually, we should probably refactor this so that the whole check happens in `samplesheets.rules` instead of a hardcoded hack, but we can do that in a further update. If it works it works.